clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -460,6 +474,7 @@ private static int parseAcks(String acksString) {
             // first make sure the metadata for the topic is available
             ClusterAndWaitTime clusterAndWaitTime = waitOnMetadata(record.topic(), record.partition(), maxBlockTimeMs);
             long remainingWaitMs = Math.max(0, maxBlockTimeMs - clusterAndWaitTime.waitedOnMetadataMs);
+            remainingWaitMs = waitOnPid(remainingWaitMs);
```

Comment:
It is passed on to the `accumulator.append`. This line just reflects that we spend more of our block time budget on waiting for the pid.